generator/config/stage/set.yaml

```
@@ -13,3 +13,61 @@ arguments:
         type:
             - expression
         variadic: object
+tests:
+    -
+        name: 'Using Two $set Stages'
+        link: 'https://www.mongodb.com/docs/manual/reference/operator/aggregation/set/#using-two--set-stages'
+        pipeline:
+            -
+                $set:
+                    totalHomework:
+                        # The $sum expression is always build as an array, even if the value is an array field name
```

```suggestion
                        # The $sum expression is always built as an array, even if the value is an array field name
```

Alternatively: "The $sum expression always builds as..."

---

Does the type of the referenced field factor in here? I assume `['$homework`]` and `$homework` would still be computed the same and follows the [$sum: Array Operand](https://www.mongodb.com/docs/manual/reference/operator/aggregation/sum/#array-operand) rules. In other words, the array wrapping has no effect other than visually being the long form API.